modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/Numeric/Numeric.es.js

```diff
@@ -253,4 +254,4 @@ const Main = ({
 Main.displayName = 'Numeric';
 
 export {Main};
-export default Main;
+export default withConfirmationField(Main);
```

Comment:
I used this [code](https://github.com/liferay/liferay-portal/blob/0c7b91e42f515549842b70fab02403a7d409942c/modules/apps/data-engine/data-engine-js-components-web/src/main/resources/META-INF/resources/js/components/loading/Loading.es.js#L33) in React as inspiration 👀 👀 .

Reply:
This is not necessarily something specific to Metal.js, but it is a pattern that is commonly used in React but was discouraged with the introduction of hooks. Just a comment not related to that but the implementation.

Just one thing to keep in mind when adding features like this, one thing I would consider an edge case is the case of issuing `onChange` with `confirmationValue`, this breaks the behavior of the core not knowing the properties of a field, for example. To edit a field property there is always a field in the related Settings Context, so a field can only edit its value or the equivalent field property, for example `label` the `name` of the field is used to decide which property should be modified.

We should always consider this to make sure that the components of the form do not know a field property or something specific, simply because we consider that the fields are dynamic and so we only offer them to the fields APIs so that it can join the core. Something that comes to mind when I see this implementation:

- Can we leave the Confirmation Field validation on the Field side without having to do an evaluation?
- Can all fields have a Confirmation Field? otherwise, I see that it is something very specific to some fields and it would be something that the core shouldn't even know about.
- Does Confirmation Field have any properties on the Settings Context side that enables it?

Well, I'm just trying to bring a few things to avoid coupling the property to the field's editing flow, if this is trivial the best thing would be to make the field change allow to say which field property you want to modify can make things easier but grow with behaviors being added in the field by emitting a lot of `onChange` to modify properties which can cause a lot of trouble and swelling.

If you want we can arrange a call to talk about it.

Reply:
Hi @matuzalemsteles, Thanks for your contribution! 

> 1. Can we leave the Confirmation Field validation on the Field side without having to do an evaluation?

We need to set the field as invalid during the evaluation to prevent the form from being submitted, just like a required field, that's why we need the confirmation value during evaluation. 

I agree with you. At first I was not happy with this solution, but it was the only way I found to have the confirmation value, when evaluating the field, without having to include the Confirmation Field in the form structure. 

It would help a lot if we could discuss this in a call.

Thanks!

Reply:
Makes sense @carolmariaabb! I think we can make a call to discuss this together and see possible changes we can make to better support it.